cmscontrib/loaders/italy_yaml.py

```
@@ -975,92 +1069,172 @@ def get_task(self, get_statement=True) -> Task | None:
-                                             conv=lambda x: "" if x is None else x)
-                output_optional_testcases = load(conf, None, "output_optional_testcases",
-                                             conv=lambda x: "" if x is None else x)
-                if len(output_only_testcases) > 0 or len(output_optional_testcases) > 0:
-                    args["task_type"] = "BatchAndOutput"
-                    output_only_codenames = set()
-                    if len(output_only_testcases) > 0:
-                        output_only_codenames = \
-                            {"%03d" % int(x.strip()) for x in output_only_testcases.split(',')}
-                        args["task_type_parameters"].append(','.join(output_only_codenames))
-                    else:
-                        args["task_type_parameters"].append("")
-                    output_codenames = set()
-                    if len(output_optional_testcases) > 0:
-                        output_codenames = \
-                            {"%03d" % int(x.strip()) for x in output_optional_testcases.split(',')}
-                    output_codenames.update(output_only_codenames)
-                    task.submission_format.extend(["output_%s.txt" % s for s in sorted(output_codenames)])
+                # Check for Communication task via manager file presence
+                paths = [os.path.join(self.path, "check", "manager"),
+                         os.path.join(self.path, "cor", "manager")]
+                is_communication = any(os.path.exists(path) for path in paths)
+                if is_communication:
+                    setup_communication()
+                else:
+                    # Default to Batch/BatchAndOutput
+                    setup_batch()
+        else:
+            # Unknown task_type - log warning and default to Batch
+            logger.warning("Unknown task_type '%s', defaulting to Batch", configured_type)
+            setup_batch()
```

Wouldn't is be better to raise an error here?